src/components/box/box.pw.tsx

```diff
@@ -0,0 +1,897 @@
+      await mount(<BoxComponentMulti flex="auto" flexGrow={value} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("flex-grow", growText);
+    });
+  });
+
+  ([
+    [10, "10"],
+    [50, "50"],
+    [100, "100"],
+  ] as [number, string][]).forEach(([value, shrinkText]) => {
+    test(`should verify flex shrink is ${value}`, async ({ mount, page }) => {
+      await mount(<BoxComponentMulti flex="auto" flexShrink={value} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("flex-shrink", shrinkText);
+    });
+  });
+
+  ["auto", "content", "fit-content", "max-content", "min-content"].forEach(
+    (basis) => {
+      test(`should verify flex basis is ${basis}`, async ({ mount, page }) => {
+        await mount(<BoxComponentMulti flexBasis={basis} />);
+        const boxElement = await getDataElementByValue(page, "box");
+        await expect(boxElement).toHaveCSS("flex-basis", basis);
+      });
+    }
+  );
+
+  [
+    "auto",
+    "baseline",
+    "left",
+    "normal",
+    "right",
+    "stretch",
+    "center",
+    "flex-start",
+    "flex-end",
+  ].forEach((justify) => {
+    test(`should verify justifySelf is ${justify}`, async ({ mount, page }) => {
+      await mount(<BoxComponentMulti justifySelf={justify} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("justify-self", justify);
+    });
+  });
+
+  [
+    "auto",
+    "baseline",
+    "normal",
+    "stretch",
+    "center",
+    "flex-start",
+    "flex-end",
+  ].forEach((align) => {
+    test(`should verify alignSelf is ${align}`, async ({ mount, page }) => {
+      await mount(<BoxComponentMulti alignSelf={align} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("align-self", align);
+    });
+  });
+
+  ([
+    [10, "10"],
+    [50, "50"],
+    [100, "100"],
+  ] as [number, string][]).forEach(([value, orderText]) => {
+    test(`should verify order is ${value}`, async ({ mount, page }) => {
+      await mount(<BoxComponentMulti order={value} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("order", orderText);
+    });
+  });
+
+  (["break-word", "anywhere"] as const).forEach((wrap) => {
+    test(`should verify overflow wrap is ${wrap}`, async ({ mount, page }) => {
+      await mount(<BoxComponentMulti overflowWrap={wrap} />);
+      const boxElement = await getDataElementByValue(page, "box");
+      await expect(boxElement).toHaveCSS("overflow-wrap", wrap);
+    });
+  });
+
+  ([
+    ["light", "rgb(102, 132, 148)", "rgb(242, 245, 246)"],
+    ["dark", "rgb(153, 173, 183)", "rgb(51, 91, 112)"],
+  ] as [BoxProps["scrollVariant"], string, string][]).forEach(
+    ([variant, thumb, track]) => {
```

Comment:
**suggestion(non-blocking)**: For parameterised tests where we are having to explicitly set types on the test parameters, what do you think about using objects for storing the parameters for each test instead of an array?

```suggestion
  [
    {variant: "light", thumbColor: "rgb(102, 132, 148)", trackColor: "rgb(242, 245, 246)"},
    {variant: "dark", thumbColor: "rgb(153, 173, 183)", trackColor: "rgb(51, 91, 112)"},
  ].forEach({variant, thumbColor, trackColor}) => {
```

That way we won't have to worry about explicitly defining the types, as the type of each parameter will be a union of constant values.

Reply:
I like this idea in principle, but given my comments above on inconsistency of using objects for some tests and arrays for other ones, I'm not so sure. Although I am more inclined to go with this than with your other suggestion whether the array was a const defined outside the tests. I'll ask about this in general I think 👍

Reply:
so it turns out that, in trying this, the type annotation is still needed, because the "variant" type is inferred as string, rather than the specific strings needed for he `scrollVariant` prop. While this can be fixed in the object version too, with type annotations/assertions or an `as const`, this seems to remove the advantage of using an object over an array here so I'm going to keep this one as an array of arrays.